src/nvim/eval.c

```diff
@@ -17409,7 +17409,8 @@ static char *save_tv_as_string(typval_T *tv, ptrdiff_t *const len, bool endnl)
   // print an error.
   if (tv->v_type != VAR_LIST) {
     const char *ret = tv_get_string_chk(tv);
-    if (ret && (*len = strlen(ret))) {
+    if (ret) {
+      *len = strlen(ret);
```

Comment:
Might be worth temporarily reverting this change to see if the problem still occurs with the new test. To avoid having to look at a bunch of failed builds, change the expect result to what happens on master.

```lua
eq("", eval('system("echo test", "")'))
```